lib/screens/regions_main/region_screen.dart

```diff
@@ -13,44 +15,53 @@ class RegionScreen extends StatefulWidget {
 class _RegionScreenState extends State<RegionScreen> {
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      body: DefaultTabController(
-        length: 3,
-        child: NestedScrollView(
-          headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
-            return <Widget>[
-              SliverAppBar(
-                actions: [
-                  TextButton(
-                    child: const Text("Join"),
-                    onPressed: () {},
-                  )
-                ],
-                expandedHeight: 200.0,
-                pinned: true,
-                flexibleSpace: FlexibleSpaceBar(
-                    centerTitle: false,
-                    title: const Text("state.regionAddress"),
-                    background: Image.network(
-                      "https://picsum.photos/400/300",
-                      fit: BoxFit.cover,
-                    )),
+    final regionId = ModalRoute.of(context)!.settings.arguments as String?;
+    return BlocProvider(
+      create: (_) => RegionBloc(regionId ?? '', regionId != null),
+      child: Scaffold(
+        body: BlocBuilder<RegionBloc, RegionState>(
+          builder: (context, state) {
+            return DefaultTabController(
+              length: 3,
+              child: NestedScrollView(
+                headerSliverBuilder: (context, isInnerBoxScrolled) {
+                  return [
+                    SliverAppBar(
+                      actions: [
+                        if (state.isBrowseView)
+                          TextButton(
+                            child: const Text("Join"),
+                            onPressed: () {},
+                          )
+                      ],
+                      expandedHeight: 200.0,
+                      pinned: true,
+                      flexibleSpace: FlexibleSpaceBar(
+                          centerTitle: false,
+                          title: const Text("state.regionAddress"),
+                          background: Image.network(
+                            "https://picsum.photos/400/300",
```

Comment:
this is mock data right ?

Reply:
oh good I forgot that

Reply:
Wait that is from main region screen haha I only did few logic changes there

Reply:
This UI is under development yet I guess